### PR TITLE
Filter parameters, returned by SSE event

### PR DESCRIPTION
**Pull request by nick863**
# Description

In the scenario when we use event handlers, we are launching actions in response to SSE events. In this case we create the `ThreadRun` object, based on the received JSON response. If the Bakend was changed and returns extra parameters, this will result in failure. In this PR we are filtering the fields, based on the Class (in this example ThreadRun)  attributes.
See Work item https://github.com/orgs/ai-platform-ml-platform/projects/7/views/2?sliceBy%5Bvalue%5D=markhiet_microsoft&pane=issue&itemId=82283437

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py [hunk 1]
```diff
@@ -600,6 +600,29 @@ async def on_unhandled_event(self, event_type: str, event_data: Any) -> None:
...
+        new_params = {}
+        valid_parameters = set(
+            filter(lambda x: not x.startswith('_') and hasattr(ThreadRun.__dict__[x], "_type"), ThreadRun.__dict__.keys())
+        )
+        for k in filter(lambda x: x in valid_parameters, parameters.keys()):
+            new_params[k] = parameters[k]
+        return new_params
+
+
 class AsyncAgentRunStream(AsyncIterator[Tuple[str, Any]]):
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py:626]: Also AgentRunStream class

File: sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py [hunk 2]
```diff
@@ -677,7 +700,9 @@ def _parse_event_data(self, event_data_str: str) -> Tuple[str, Any]:
             AgentStreamEvent.THREAD_RUN_CANCELLED,
             AgentStreamEvent.THREAD_RUN_EXPIRED,
         }:
-            event_data_obj = ThreadRun(**parsed_data) if isinstance(parsed_data, dict) else parsed_data
+            ParamCorrector
+            event_data_obj = ThreadRun(**ParamCorrector.filter_parameters(
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py:704]: We probably need to check all types, like RunStep, ThreadMessage aswell

Comment (nick863) [reply sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py]: Fixed.

File: sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py (hunk 1 above)
Comment (nick863) [reply sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py:640]: Fixed

File: sdk/ai/azure-ai-client/tests/endpoints/unit_tests.py [hunk 3]
```diff
@@ -112,3 +117,78 @@ def test_sas_token_credential_class_real(self, **kwargs):
...
+                'role': 'assistant',
+                'content': 'Test',
+                'assistant_id': '9911',
+                'run_id': '11',
+                'attachments': ['4', '8', '15', '16', '23', '42'],
+                'metadata': {'foo', 'bar'}
+            }, ThreadMessage)
+        ]
+    )
+    def test_correct_thread_params(self, valid_params, model_cls):
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/tests/endpoints/unit_tests.py:172]: Thank you for adding test!

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py [hunk 4]
```diff
@@ -600,6 +600,43 @@ async def on_unhandled_event(self, event_type: str, event_data: Any) -> None:
         pass
 
 
+class ParamCorrector():
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py:603]: sounds good

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (nick863) [reply sdk/ai/azure-ai-client/azure/ai/client/models/_patch.py]: I have moved out the utility methods from the class.

**Review by dargilco — APPROVED** [review]

**Pull request merged**